src/main/java/de/tum/in/www1/artemis/service/connectors/localci/scaparser/strategy/PMDParser.java

```diff
@@ -1,71 +1,77 @@
+class PMDParser implements ParserStrategy {
 
-    private static final String VIOLATION_ATT_ENDCOLUMN = "endcolumn";
+    private final XmlMapper xmlMapper = new XmlMapper();
 
     @Override
-    public StaticCodeAnalysisReportDTO parse(Document doc) {
-        StaticCodeAnalysisReportDTO report = new StaticCodeAnalysisReportDTO();
-        report.setTool(StaticCodeAnalysisTool.PMD);
+    public StaticCodeAnalysisReportDTO parse(String xmlContent) {
+        try {
+            PMDReport pmdReport = xmlMapper.readValue(xmlContent, PMDReport.class);
+            return createReportFromPMDReport(pmdReport);
+        }
+        catch (IOException e) {
+            throw new RuntimeException("Failed to parse XML", e);
+        }
+    }
+
+    private StaticCodeAnalysisReportDTO createReportFromPMDReport(PMDReport pmdReport) {
         List<StaticCodeAnalysisIssue> issues = new ArrayList<>();
-        Element root = doc.getDocumentElement();
-
-        // Iterate over all <file> elements
-        for (Element fileElement : getChildElements(root, FILE_TAG)) {
-            // Extract the file path
-            String unixPath = ParserUtils.transformToUnixPath(fileElement.getAttribute(FILE_ATT_NAME));
-
-            // Iterate over all <violation> elements
-            for (Element violationElement : getChildElements(fileElement)) {
-                StaticCodeAnalysisIssue issue = new StaticCodeAnalysisIssue();
-                issue.setFilePath(unixPath);
-
-                issue.setRule(violationElement.getAttribute(VIOLATION_ATT_RULE));
-                issue.setCategory(violationElement.getAttribute(VIOLATION_ATT_RULESET));
-                issue.setPriority(violationElement.getAttribute(VIOLATION_ATT_PRIORITY));
-                issue.setStartLine(ParserUtils.extractInt(violationElement, VIOLATION_ATT_BEGINLINE));
-                issue.setEndLine(ParserUtils.extractInt(violationElement, VIOLATION_ATT_ENDLINE));
-                issue.setStartColumn(ParserUtils.extractInt(violationElement, VIOLATION_ATT_BEGINCOLUMN));
-                issue.setEndColumn(ParserUtils.extractInt(violationElement, VIOLATION_ATT_ENDCOLUMN));
-                issue.setMessage(ParserUtils.stripNewLinesAndWhitespace(violationElement.getTextContent()));
+        StaticCodeAnalysisReportDTO report = new StaticCodeAnalysisReportDTO(StaticCodeAnalysisTool.PMD, issues);
+
+        for (FileViolation fileViolation : pmdReport.files()) {
+            String unixPath = ParserStrategy.transformToUnixPath(fileViolation.fileName());
 
+            for (Violation violation : fileViolation.violations()) {
+                StaticCodeAnalysisIssue issue = new StaticCodeAnalysisIssue(unixPath, violation.beginLine(), violation.endLine(), violation.beginColumn(), violation.endColumn(),
+                        violation.rule(), violation.ruleset(), violation.message().strip(), violation.priority(), null  // Assuming there is no penalty info in PMD XML
```

Comment:
The comment about no penalty is different for the other two tools. Is that on purpose?
I like this version the most, if they are all supposed to mean the same thing